shared/src/main/java/com/couchbase/lite/Authenticator.java

```diff
@@ -0,0 +1,15 @@
+package com.couchbase.lite;
+
+import java.util.Map;
+
+/**
+ * Authenticator is an opaque based authenticator interface and not intended for application to
+ * implement a custom authenticator by subclassing Authenticator interface.
+ */
+public abstract class Authenticator {
+    static final String kCBLReplicatorAuthOption = "auth";
+    static final String kCBLReplicatorAuthUserName = "username";
+    static final String kCBLReplicatorAuthPassword = "password";
```

Comment:
Currently `C4Socket` is implemented in `litecore` layer. It should be in `lite` layer. I'd like to keep these constants in Autheticator till `C4Socket` is moved to `lite` layer.